algorithms/src/polycommit/kzg10/data_structures.rs

```diff
@@ -131,6 +131,8 @@ impl<E: PairingEngine> FromBytes for UniversalParams<E> {
     }
 }
 
+// It is safe to cast `supported_degree_bounds.len()` and individual bounds to a `u32` because they are read and used as u32
+#[allow(clippy::cast_possible_truncation)]
```

Comment:
This should be fixed, not allowed. See https://github.com/AleoHQ/snarkVM/blob/testnet3/synthesizer/coinbase/src/helpers/coinbase_solution/bytes.rs#L37 for how to proceed